### PR TITLE
Hotfix: Show user id for direct chat case in search

### PR DESCRIPTION
**Pull request by nqhhdev**
- Issue: 
  - Appbar overlay in chat detail view when clicking on app bar in chat group
  - Information widget (Direct chat) in search is wrongly 
  
- Resolved:

https://github.com/linagora/twake-on-matrix/assets/99852347/6066f533-86e1-4dee-b251-ef9ff3d6bf39

Simulator Screenshot - iPhone 14 Pro Max - 2023-08-15 at 23 14 39

Simulator Screenshot - iPhone 14 Pro Max - 2023-08-15 at 23 14 48

Simulator Screenshot - iPhone 14 Pro Max - 2023-08-15 at 23 14 50

Simulator Screenshot - iPhone 14 Pro Max - 2023-08-15 at 23 14 53

Simulator Screenshot - iPhone 14 Pro Max - 2023-08-15 at 23 14 56

Simulator Screenshot - iPhone 14 Pro Max - 2023-08-15 at 23 29 29

Simulator Screenshot - iPhone 14 Pro Max - 2023-08-15 at 23 29 35

Simulator Screenshot - iPhone 14 Pro Max - 2023-08-15 at 23 29 38

Simulator Screenshot - iPhone 14 Pro Max - 2023-08-15 at 23 39 37

### CODE REVIEW

**Review by hoangdat — APPROVED** [review]

**Review by sherlockvn — APPROVED** [review]

File: lib/pages/chat/chat_view.dart [hunk 1]
```diff
@@ -110,6 +110,7 @@ class ChatView extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
+    final chatViewStyle = ChatViewStyle();
```
Comment (hoangdat) [inline lib/pages/chat/chat_view.dart:113]: no need instance of `style`

File: lib/pages/chat_details/chat_details_view.dart [hunk 2]
```diff
@@ -48,33 +49,49 @@ class ChatDetailsView extends StatelessWidget {
...
-                      onPressed: () => FluffyShare.share(
-                        AppConfig.inviteLinkPrefix + room.canonicalAlias,
-                        context,
+                title: Padding(
+                  padding: ChatDetailViewStyle.paddingTitle(context),
+                  child: Row(
+                    children: [
+                      IconButton(
+                        icon: const Icon(Icons.close_outlined),
+                        onPressed: () => GoRouterState.of(context)
```
Comment (drminh2807) [inline lib/pages/chat_details/chat_details_view.dart:64]: move it to ChatDetailsController

File: lib/pages/forward/forward_view.dart [hunk 3]
```diff
@@ -28,6 +28,8 @@ class _ForwardViewState extends State<ForwardView> {
   bool isShowRecentlyChats = true;
   bool isSearchBarShow = false;
 
+  final forwardViewStyle = ForwardViewStyle();
```
Comment (drminh2807) [inline lib/pages/forward/forward_view.dart:31]: dead code ?

File: lib/pages/chat_details/chat_details.dart [hunk 4]
```diff
@@ -29,6 +31,10 @@ class ChatDetailsController extends State<ChatDetails> {
   List<User>? members;
   bool displaySettings = false;
 
+  final responsive = getIt.get<ResponsiveUtils>();
+
+  bool get isMobile => responsive.isMobile(context);
+
```
Comment (hoangdat) [inline lib/pages/chat_details/chat_details.dart:37]: not use :)))

File: lib/pages/chat_list/chat_list.dart [hunk 5]
```diff
@@ -75,6 +76,10 @@ class ChatListController extends State<ChatList>
 
   final _getRecoveryWordsInteractor = getIt.get<GetRecoveryWordsInteractor>();
 
+  final responsive = getIt.get<ResponsiveUtils>();
+
+  bool get isDesktop => responsive.isDesktop(context);
+
```
Comment (hoangdat) [inline lib/pages/chat_list/chat_list.dart:82]: idem

File: lib/pages/search/recent_item_widget.dart [hunk 6]
```diff
@@ -14,11 +15,13 @@ import 'package:matrix/matrix.dart';
 class RecentItemWidget extends StatelessWidget {
   final PresentationSearch presentationSearch;
   final String highlightKeyword;
+  final SearchController searchController;
```
Comment (hoangdat) [inline lib/pages/search/recent_item_widget.dart:18]: why need this controller?

File: lib/widgets/twake_components/twake_header.dart [hunk 7]
```diff
@@ -15,20 +14,18 @@ class TwakeHeader extends StatelessWidget
...
   @override
   Widget build(BuildContext context) {
-    final isWebWidth = ResponsiveUtils().isDesktop(context);
-
     return AppBar(
       toolbarHeight: TwakeHeaderStyle.toolbarHeight,
       automaticallyImplyLeading: false,
       leadingWidth: TwakeHeaderStyle.leadingWidth,
-      leading: !isWebWidth
+      leading: !controller.isDesktop
```
Comment (hoangdat) [inline lib/widgets/twake_components/twake_header.dart:21]: use the style instead

**Review by drminh2807 — APPROVED** [review]

File: lib/pages/search/search.dart [hunk 8]
```diff
@@ -159,6 +159,20 @@ class SearchController extends State<Search>
...
+
+    if (roomId != null) {
+      final room = Matrix.of(context).client.getRoomById(roomId);
+      final userId =
+          room?.unsafeGetUserFromMemoryOrFallback(directChatMatrixID).id;
+      return userId;
+    } else {
+      return null;
+    }
+  }
```
Comment (hoangdat) [inline lib/pages/search/search.dart:174]: what use for?

**Review by hoangdat — APPROVED** [review]

**Pull request merged**